draft-ietf-webpush-vapid.md

```diff
@@ -202,6 +216,40 @@ application server would not be required to accept and store cookies, though the
 push service would not be able to offload any state as a result.
 
 
+## Subscription Association
+
+The public key or otherwise public token offered by the application service to
+the push service may also be used to associate a subscription with the
+application server that can prove possession of the private counterpart.
+
+Subscription association reduces the reliance on endpoint secrecy by requiring
+this proof to be presented when requesting delivery of a push message.  This can
+provide an additional level of security in a situation where the application
+server may suffer a data leak.
+
+### Amendments to Subscribing for Push Messages
+
+The user agent may include the public token of the application server to
+associate the subscription with when creating a subscription, when developer has
+made the association token available.  For example, the Web Push API [API] could
+enable the public token to be specified in the "association" field of the
+PushSubscriptionOptions dictionary.
```

Comment:
This is a good point, it would enable subscriptions to be associated with multiple application servers. Let me include it as an option to consider.

We actually used to support this, but have removed it since. I'll find out why.